vespalib/src/tests/component/component.cpp

```diff
@@ -197,6 +198,79 @@ TEST("testSimple")
     EXPECT_TRUE(vs.matches(v));
 }
 
+TEST("testZeros")
```

Comment:
Consider increasing the verbosity of the test name so that it states _what_ property of zeroes it expects to hold. E.g. something like `"Version toAbbreviatedString() truncates trailing zeroed components while toString() does not"`.

Also consider factoring out `VersionSpecification` specific parts to its own test case. This also means the `// test Foo:` comments can be removed without any information loss. Other than that, great test coverage!